src/screens/welcome/install-referrer/onboardingPartner.ts

```diff
@@ -0,0 +1,127 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { Platform } from 'react-native'
+
+import Minilog from '@cozy/minilog'
+
+import { getInstallReferrer } from './androidPlayInstallReferrer'
+
+import strings from '/constants/strings.json'
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+
+const log = Minilog('Referral')
+
+const ONBOARDING_PARTNER_CAMPAIGN = 'onboarding_partner'
+
+export const NO_ONBOARDING_PARTNER = 'NO_ONBOARDING_PARTNER'
+
+interface NoOnboardingPartner {
+  hasReferral: false
+}
+
+interface WithOnboardingPartner {
+  source: string
+  context: string
+  hasReferral: true
+}
+
+export type OnboardingPartner = WithOnboardingPartner | NoOnboardingPartner
+
+const noOnboardingPartner = (): NoOnboardingPartner => ({ hasReferral: false })
+
+const extractOnboardingPartner = (
+  installReferrer: string
+): OnboardingPartner | null => {
+  const url = new URL(`https://cozy.io?${installReferrer}`)
```

Comment:
I don't understand this domain? Don't we have to only generate links as : play.google.com/..?utm_... ? What is this `cozy.io` stuff?

Reply:
The used API returns only the query string in the form of `utm_source=SOME_SOURCE&utm_content=SOME_CONTEXT&utm_campaign=onboarding_partner&anid=admob`

The `cozy.io` part is just some whatever domain so I can create an URL (I don't know how to create a `new URLSearchParams` from a string)